src/libopensc/pkcs15-pin.c

```diff
@@ -367,6 +367,7 @@ int sc_pkcs15_verify_pin(struct sc_pkcs15_card *p15card,
 	r = sc_pin_cmd(card, &data, &auth_info->tries_left);
 	sc_log(ctx, "PIN cmd result %i", r);
 	if (r == SC_SUCCESS)
+	if (r == SC_SUCCESS && pin_obj->usage_counter == 0)
```

Comment:
"Use PIN caching" is a workaround for other pkcs11 tools that are not compliant, since I'm only using OpenSC code, I don't see how the "Use PIN caching" conf is related to the bug, to my fix or to @viktorTarasov pointing that the bug can be somewhere else.